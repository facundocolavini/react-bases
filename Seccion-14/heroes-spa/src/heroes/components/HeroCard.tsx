import { Link } from 'react-router-dom';
import { Hero } from "../models"

interface HeroCard extends Hero {
    
}

const CharactersByHero = ({alter_ego, characters}: any) =>{
  // if (alter_ego === characters) return (<></>);
  // return  <p>{ characters }</p>   
  return (alter_ego === characters) ? <></>: <p>{ characters }</p>
}
export const HeroCard = ({ 
    id,
    superhero,
    publisher,
    alter_ego,
    first_appearance,
    characters } : HeroCard) => {

  // const charactersByHero  = (<p>{ characters }</p>);   
  const  heroImageUrl = `./assets/heroes/${id}.jpg`;

  return (
    <div className="col animate__animated animate__fadeIn p-1 mt-2">
      <div className="card" key={id}>
        <div className="row">
          <div className="col-4">
             <img src={heroImageUrl} alt={ superhero } className="card-img" />
          </div>
          <div className="col-8">
            <div className="card-body">
              <h5 className="card-title">{ superhero }</h5>
              <p className="card-text">{ alter_ego }</p>
              {/* {
                (alter_ego !== characters ) && charactersByHero
              } */}
              <CharactersByHero characters={ characters } alter_ego= { alter_ego }/> 
              <p className="card-text">
                <small className="text-muted">{ first_appearance }</small>
              </p>
              <Link to={`/hero/${id}`}>
                MÃ¡s..
              </Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}