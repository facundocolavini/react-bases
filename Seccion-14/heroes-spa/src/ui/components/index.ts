export * from './Navbar';
export * from './AlertsTypes';

