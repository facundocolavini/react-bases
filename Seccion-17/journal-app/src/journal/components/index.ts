export * from './NavBar';
export * from './SideBar';
export * from './SideBarListItems';
export * from './ImageGallery';
