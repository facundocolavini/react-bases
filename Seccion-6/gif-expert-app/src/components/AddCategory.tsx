import { useState } from 'react';
import PropTypes from 'prop-types';

type Props = {
    onNewCategory: (newCategory: string) => void,
}

export const AddCategory = ({onNewCategory}: Props): JSX.Element => {
   
    const [inputValue, setInputValue] = useState<string>('') 

    /* Handlers */
    const onHandleChange = ({target} :React.ChangeEvent<HTMLInputElement>):void=>{
        setInputValue( target.value );
    }

    const submitForm = (e: React.FormEvent) =>{
        e.preventDefault()
        const newInputValue = inputValue.trim() 
        if( newInputValue.length <= 1) return;
        onNewCategory(newInputValue)
        setInputValue('')
    }
    
  return (    
    <form onSubmit={ submitForm } aria-label="form">
        <input 
            type="text"
            value={inputValue}
            onChange={ onHandleChange }
            placeholder="Buscar gifs"
        />
    </form>
  )
};

AddCategory.propTypes = {
    onNewCategory : PropTypes.func.isRequired 
}
