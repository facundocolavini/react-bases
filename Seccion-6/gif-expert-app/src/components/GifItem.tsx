import React from 'react'
import PropTypes from 'prop-types'

type Gif = {
    id?: string;
    title? : string;
    url? : string;
}

export const GifItem = ({ title, url } : Gif) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
}