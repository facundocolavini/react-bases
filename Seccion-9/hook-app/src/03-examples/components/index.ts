export * from './LoadingQuotes';
export * from './Quotes';
export * from './ErrorQuotes';