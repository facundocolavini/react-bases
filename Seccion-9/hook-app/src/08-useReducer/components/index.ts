export { default as TodoAdd } from './TodoAdd';
export { default as TodoAddSimple } from './TodoAddSimple';
export * from './TodoItem';
export { default as TodoList } from './TodoList';
