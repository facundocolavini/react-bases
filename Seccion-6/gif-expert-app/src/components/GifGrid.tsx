import { useFetchGifs } from "../hooks/useFetchGifs"
import { GifItem } from "./GifItem"


type Props = {
    category: string,
}

type Gif = {
  id?: string;
  title : string;
  url : string;
}

export const GifGrid = ({ category }: Props): JSX.Element => {
  const {images, isLoading } = useFetchGifs( category )

  return (
    <>
      <h3>{category}</h3>  
      {isLoading }
      {isLoading && <h2> Cargando...</h2>}
      <div className="card-grid">
        
        {/* Tarea colocar dinamicamente el titulo de cada gif */}
        {
          images.map((image: Gif) =>(
            <GifItem  key={image.id}{...image} />
          ))
        }
      </div>
    </>
  )
}
