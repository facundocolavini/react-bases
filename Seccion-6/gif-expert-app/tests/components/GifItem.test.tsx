
import * as React from 'react'
import { describe, expect, test } from '@jest/globals';
import { fireEvent, render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem'

/*
Cosas a probar  en el componente GifItem
    1. a√±adir propTypes 
        a. title obligatorio 
        b. url obligatorio 
        c. id es opcional (de momento)

    2. Probar sin las propsTypes 
        a. hacer prueba si no se le envia el titulo
        b. hacer una prueba si no se le envia url

    3. Evaluar el snapshot

*/ 

describe('Trabajando en el componente <GifItem/>', () => {
    const title: string = 'Saitama';
    const url: string = 'http://localhost:8080/';
    

   
    test('Debe de hacer match con el snapshot',()=>{
       const{ container } = render(<GifItem title={title} url={url} />)

       // Evaluacion 
       expect( container ).toMatchSnapshot();
    })

    test('Debe de mostrar la imagen con el URL y el ALT indicado', ()=>{
        render(<GifItem title={title} url={url} />)
        
        // Obtener un atributo de mi elemento  
        // expect( screen.getByRole('img').getAttribute('src')?.valueOf()).toBe( url );
        // expect( screen.getByRole('img').getAttribute('alt')?.valueOf()).toBe( title );

        // otra forma de evaluar los atributos de un elemento
        const {src , alt}: HTMLImageElement = screen.getByRole('img');
        
        expect( src ).toBe( url);
        expect( alt ).toBe( alt );


        screen.debug();
    })

    test('Debe mostrar el titulo en el componente', ()=>{
        render(<GifItem title={title} url={url} />);
        expect( screen.getByText( title )).toBeTruthy();
    })
   
})
