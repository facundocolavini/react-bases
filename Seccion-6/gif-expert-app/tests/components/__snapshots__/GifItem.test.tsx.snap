// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trabajando en el componente <GifItem/> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://localhost:8080/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
